### ISSUE TITLE
regularization not working

### ISSUE THREAD

**Issue by chaohuang** [keras-team/keras#9296]
I'm fine-tuning a VGG16 net, and the training accuracy is around 97% and the validation accuracy is only about 75%, so I try to add some regularization in the net.

I changed the VGG16 Keras codes and added L2-norm regularization in each convolution layer. I also added the same regularization in the dense layers, which were created by myself. In addition, I added dropout layers between dense layers. Here are the codes:

```
x = Conv2D(512, (3, 3), activation='relu', padding='same', name='block4_conv3', kernel_regularizer=regularizers.l2(weight_decay))(x)
...
x = Dense(1024, activation='relu', kernel_regularizer=regularizers.l2(weight_decay))(x)
```

However, the regularization doesn't work so far. Even I set weight_decay = 10^9, the loss and accuracy are almost the same as those without regularization.

What could possibly go wrong?
